GiniVision/Classes/Core/Screens/Camera/CameraPreviewViewController.swift

```diff
@@ -0,0 +1,292 @@
+            cameraState = .valid
+            previewView.session = validCamera.session
+            NotificationCenter.default.addObserver(self,
+                                                   selector: #selector(subjectAreaDidChange),
+                                                   name: NSNotification.Name.AVCaptureDeviceSubjectAreaDidChange,
+                                                   object: camera?.videoDeviceInput?.device)
+        }
+        previewView.drawGuides(withColor: giniConfiguration.cameraPreviewCornerGuidesColor)
+        
+        view.insertSubview(previewView, at: 0)
+        Constraints.pin(view: previewView, toSuperView: view)
+
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        updatePreviewViewOrientation() // Video orientation should be updated once the view has been loaded
+    }
+    
+    public override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        camera?.start()
+    }
+    
+    public override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        camera?.stop()
+    }
+    
+    public override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        
+        coordinator.animate(alongsideTransition: { [weak self] _ in
+            guard let `self` = self else {
+                return
+            }
+            
+            self.updatePreviewViewOrientation()
+        })
+    }
+    
+    fileprivate func updatePreviewViewOrientation() {
+        let orientation: AVCaptureVideoOrientation
+        if UIDevice.current.isIpad {
+            orientation =  interfaceOrientationsMapping[UIApplication.shared.statusBarOrientation] ?? .portrait
+        } else {
+            orientation = .portrait
+        }
+        
+        let previewLayer = (self.previewView.layer as? AVCaptureVideoPreviewLayer)
+        previewLayer?.connection?.videoOrientation = orientation
+    }
+    
+    fileprivate func setupCamera(giniConfiguration: GiniConfiguration) {
+        camera = Camera(giniConfiguration: giniConfiguration) { error in
+            if let error = error {
+                switch error {
+                case .notAuthorizedToUseDevice:
+                    addNotAuthorizedView()
+                default:
+                    if GiniConfiguration.DEBUG {
+                        cameraState = .valid
+                        #if targetEnvironment(simulator)
+                        addDefaultImage()
+                        #endif
+                    }
+                }
+            }
+        }
+                
+        if giniConfiguration.qrCodeScanningEnabled {
+            camera?.setupQRScanningOutput()
+            camera?.didDetectQR = { [weak self] qrDocument in
+                guard let `self` = self else { return }
```

Comment:
Same here.

Reply:
Not here, since the delegate needs it in the line below

Reply:
Why can't you write `self?.delegate?.cameraPreview(self, didDetect: qrDocument)`